src/plugins/contextMenu/predefinedItems/clearColumn.js

```diff
@@ -26,11 +27,12 @@ export default function clearColumnItem() {
       if (!selected) {
         return true;
       }
-      const [startRow, startColumn, endRow] = selected[0];
-      const entireRowSelection = [startRow, 0, endRow, this.countCols() - 1];
-      const rowSelected = entireRowSelection.join(',') === selected.join(',');
 
-      return startColumn < 0 || this.countCols() >= this.getSettings().maxCols || rowSelected;
+      const startColumn = selected[0][1];
+
+      return startColumn < 0
+        || this.countCols() >= this.getSettings().maxCols
+        || !this.selection.isEntireColumnSelected();
```

Comment:
```suggestion
      return !this.selection.isSelectedByColumnHeader();
```
Maybe this could be simplified to something like this?